### PR TITLE
test: dismiss quit_more from Lua

### LINKED ISSUE #11224: lua/python/…: "q" can not quit the pager
- `nvim --version`: v0.5.0-68-gfe9d54f41
- `vim -u DEFAULTS` (version: ) behaves differently?
- Operating system/version: macos
- Terminal name/version: alacritty
- `$TERM`:

### Steps to reproduce using `nvim -u NORC`

1. nvim -u NONE
2. :lua print(vim.inspect(vim))
3. `q` can not quit the pager

### Expected behaviour
`q` should quit the pager

### PR DESCRIPTION
**Pull request by blueyed**
Fixes https://github.com/neovim/neovim/issues/11224.

Discussion for Vim: https://github.com/vim/vim/pull/5053

### CODE REVIEW

**Review by justinmk — APPROVED** [review]
Let's try it, no need to wait on upstream.

**Comment by blueyed** [comment]
Related Vim patch: vim/vim@b98678a (v8.1.2179).

**Comment by justinmk** [comment]
Maybe rebase after https://github.com/neovim/neovim/pull/11225 ?

still WIP? Let's mark `vim-patch:8.1.2179` with this also.

**Comment by blueyed** [comment]
@justinmk 
Yeah, I'll have to look into this, and will rebase it then also.

**Review by 88maomao — APPROVED** [review]

**Comment by janlazo** [comment]
Patch v8.1.2179 is N/A in https://github.com/neovim/neovim/pull/11160.

File: src/nvim/message.c [hunk 1]
```diff
@@ -264,6 +264,10 @@ bool msg_attr_keep(char_u *s, int attr, bool keep, bool multiline)
   int retval;
   char_u *buf = NULL;
 
+  if (quit_more) {
```
Comment (bfredl) [inline src/nvim/message.c:267]: don't we still want to add it to message history if applicable?

Comment (88maomao) [reply src/nvim/message.c:267]: no. we did not bother reading long output till the end and pressed `q`

**Comment by justinmk** [comment]
> Fixes https://github.com/neovim/neovim/issues/11224

closed by https://github.com/neovim/neovim/pull/16537 .

But this PR adds a test, that is probably worth having.

**Comment by blueyed** [comment]
I won't be able to work on this myself, so please feel free to pull it out from here.

**Pull request merged**